From 067aacc2dc8ab1b02286693656df801e08dec49e Mon Sep 17 00:00:00 2001
From: Niko Tyni <ntyni@debian.org>
Date: Thu, 14 Aug 2014 21:50:00 +0000
Subject: [PATCH 1/3] Fix Wx::_load_plugin() segfaulting when
 wxPluginManager::LoadLibrary fails

It would be nice to also signal an error message upwards in the caller
stack, but I can't see an easy way to set dl_last_error in DynaLoader
from here.

Bug-Debian: https://bugs.debian.org/758127
---
 Wx.xs | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/Wx.xs b/Wx.xs
index 01029fc..3f5dede 100644
--- a/Wx.xs
+++ b/Wx.xs
@@ -422,7 +422,10 @@ _load_plugin( string, int flags = 0 /* to be compatible with dl_load_file */ )
 #endif
 #endif
     wxDynamicLibrary *lib = wxPluginManager::LoadLibrary( string, wxDL_VERBATIM );
-    RETVAL = PTR2IV( lib->GetLibHandle() );
+    if (lib)
+        RETVAL = PTR2IV( lib->GetLibHandle() );
+    else
+        RETVAL = 0;
   OUTPUT:
     RETVAL
 
-- 
2.1.0.rc1

